[FEATURE]#14 디자인 시스템 1차 개발

jiwon2724 opened the pull request:
## 참고사항 
- `+` / `-` 합산 300을 유지하기 위해 디자인 시스템을 나눠서 개발하겠습니다!
- 추가되어야 하는 기능이나 수정되어야 하는점 코멘트 남겨주시면 감사하겠습니다~
- Button의 디자인 시스템 문서를 작성해봤어요! 참고해서 봐주시면 감사하겠습니다😎
https://www.notion.so/yapp-workspace/Button-5aa4cf0035754bf5879560f46c70fbb1?pvs=4

![image](https://github.com/YAPP-Github/22nd-Android-Team-1-Android/assets/70135188/6f7fbc3c-ce16-4bf9-afd1-5e28479aaaec)


## 내용
1. 디자인 시스템 1차 개발 완료
2. font, color, 아이콘 설정 완료
3. 커스텀 Button 구현 완료 
### Done!
 - [x] font 설정
 - [x] color 설정
 - [x] Button 커스텀
 - [x] 아이콘 저장

### 관련 Issue
#14

jiwon2724 committed:
[BASE]#14 - colors.xml 설정

jiwon2724 committed:
[BASE]#14 - 폰트파일 설정

jiwon2724 committed:
[BASE]#14 - core-ui 모듈 패키지 생성 및 이동

jiwon2724 committed:
[BASE]#14 - font 설정 완료

jiwon2724 committed:
[BASE]#14 - icons 저장 완료

jiwon2724 committed:
[BASE]#14 - dev 브랜치 적용 안된부분 추가

jiwon2724 committed:
[FEAT]#14 - 디자인 시스템 Button 구현 완료
- ViewExtensions.kt visible, invisible, gone 확장함수 추가
- dimens, attrs, 속성 추가
- Button CustomView에 필요한 요소 추가

jiwon2724 committed:
[REFACTOR] - MediumButton thumbBackgroundTint 추가

myung6024 commented on core/ui/src/main/res/layout/peonlee_large_button.xml:
```diff
@@ -0,0 +1,35 @@
+        app:layout_constraintTop_toTopOf="parent">
+    </TextView>
+
+    <ImageView
+        android:id="@+id/iv_chevron_right"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginStart="@dimen/margin_4dp"
+        android:background="@drawable/icons_chevron_right"
+        app:layout_constraintBottom_toBottomOf="@+id/tv_title"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintHorizontal_bias="0.5"
+        app:layout_constraintStart_toEndOf="@+id/tv_title"
+        app:layout_constraintTop_toTopOf="@+id/tv_title">
+    </ImageView>
```
tv_title이 엄청 길어졌을 때에는 어떻게 체인이 어떻게 걸릴까요?

jiwon2724 replied:
tv_title의 lineCount가 2줄 이상인 경우 의도한대로 UI가 그려지지 않네요😭
2줄 이상인 케이스에 대해서 말씀하신 부분이 맞으실까요~?.?

myung6024 replied:
넵 뭔가 오른쪽에 이미지와 연결이 되어있는데 텍스트가 길어지면 이미지가 화면을 넘어갈 것 같아서요

myung6024 replied:
그런데 이거 관련 스펙이 없어서 고민이긴 하네요. 버튼 텍스트가 그렇게 길어질까 싶기도 하고...

jiwon2724 replied:
음 일단 onMeasure에서 tv_title의 lineCount로 분기하긴 했는데, 불필요하면 지워도 괜찮을 것 같습니다 ㅎㅎ

myung6024 replied:
요 부분은 사용처 보고 나중애 처리해도 될 것 같습니다

jiwon2724 replied:
네 확인했습니다~~😎

myung6024 commented on core/ui/src/main/res/layout/peonlee_large_button.xml:
```diff
@@ -0,0 +1,35 @@
+    android:layout_height="@dimen/button_height_large">
+
+    <TextView
+        android:id="@+id/tv_title"
```
저희 컨벤션이 카멜 케이스였던 것 같은데..!

jiwon2724 replied:
XML은 스네이크 케이스로 되어 있어서 적용했습니다! 혹시 제가 다르게 알고 있는 부분이 있을까요~?.?

aurora32s replied:
저는 tv_[이름]에서
이름 부분이 첫 문제는 소문자로 그 이후 단어는 카멜 케이스로 표시하는 걸로 알고 있어요.🤔

jiwon2724 replied:
아아 이해했습니다!!ㅎㅎ 수정해서 반영하겠습니답!!🙏

myung6024 commented on core/ui/src/main/res/layout/peonlee_large_button.xml:
```diff
@@ -0,0 +1,35 @@
+        android:background="@drawable/icons_chevron_right"
+        app:layout_constraintBottom_toBottomOf="@+id/tv_title"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintHorizontal_bias="0.5"
```
layout_constraintHorizontal_bias가 양쪽에 들어가야 하나요?

jiwon2724 replied:
아니요! 불필요합니다! 제약 설정할 때 잘못 들어간 것 같네요..ㅎㅎ 수정하겠습니다!

myung6024 commented on core/ui/src/main/res/layout/peonlee_medium_button.xml:
```diff
@@ -0,0 +1,34 @@
+    android:layout_height="@dimen/button_height_medium">
+
+    <TextView
+        tools:text="test text"
```
요거 줄 정렬하면 tools:text가 가장 하단으로 내려갈 것 같은데 정렬 한번 부탁드립니다~!

jiwon2724 replied:
넵!!!! 신경쓰겠습니다...!!!!!🙏🙏🙏🙏🙏🙏🙏

myung6024 commented on core/ui/src/main/java/com/peonlee/core/ui/extensions/ViewExtensions.kt:
```diff
@@ -0,0 +1,15 @@
+
+import android.view.View
+
+fun View.gone() {
+    this.visibility = View.GONE
+}
+
+fun View.inVisible() {
+    this.visibility = View.INVISIBLE
+}
+
+fun View.visible() {
+    this.visibility = View.VISIBLE
+}
```
```
public inline var View.isInvisible: Boolean
    get() = visibility == View.INVISIBLE
    set(value) {
        visibility = if (value) View.INVISIBLE else View.VISIBLE
    }

public inline var View.isGone: Boolean
    get() = visibility == View.GONE
    set(value) {
        visibility = if (value) View.GONE else View.VISIBLE
    }
```

요런 extension이 있어서 중복되는 느낌이 들긴하네요!

jiwon2724 replied:
네 확인했습니다! `isXXX`를 사용하면 불필요 할 것 같아요! 삭제하구 반영하겠습니다~

myung6024 commented on core/ui/src/main/java/com/peonlee/core/ui/designsystem/button/MediumButton.kt:
```diff
@@ -0,0 +1,111 @@
+        thumbsBackground: Int,
+        thumbsBackgroundTint: Int
+    ) {
+        if (isShowingThumbs) binding.ivThumbs.visible() else binding.ivThumbs.gone()
```
`binding.ivThumbs.isVisible = isShowingThumbs`

이렇게 if 안쓰고 적용할 수도 있어요!

jiwon2724 replied:
간결하구 가독성 면에서도 더 좋은 것 같아요!👍 감사합니다 ㅎㅎ 적용해서 반영할게요!! 천재만재!!😎

aurora32s replied:
LargeButton에도 있어요. :-)

aurora32s commented on core/ui/src/main/res/values/themes.xml:
```diff
@@ -16,6 +16,86 @@
     <style name="Theme.MyApplication" parent="android:Theme.Material.Light.NoActionBar" />
-</resources>
+
+    <style name="PeonLee.Typography" parent="android:TextAppearance">
```
[질문] Typography 설정을 할 때 lineHeight은 별도로 해줄 필요가 없는걸까요?

jiwon2724 replied:
lineHeight에 대해서 명준님이랑 뎊캡 때 간단하게 폰트 관련해서 리뷰했는데, 적용을 지양하는 방향으로 얘기 되어서
일단은 넣지 않았어요! 오늘 스크럼 때 이 부분 다시 얘기해 보면 좋을 것 같습니답 ㅎㅎ🙌

myung6024 replied:
안드로이드 xml에서 뜻하는 lineHeight와 디자인 lineHeight의 의미가 조금 다릅니다. 그런데 또 compose 최신 버전에서는 lineHeight를 제대로 적용할 수 있도록 지원하더라구요.

jiwon2724 committed:
[REFACTOR]#14 - comment 반영
- LargeButton, MediumButton onMeasure 측정
- ViewExtensions.kt 삭제
- 미사용 bias 속성 삭제
- xml 코드 정렬
- isVisible 속성 활용

aurora32s reviewed (commented):
지원님 :-) 디자인 시스템 관련 1차 개발 하시느라 수고하셨습니다.🤗
xml 환경에서 customView를 만드는 방법에 대해 저도 배워갑니다!

ps. xml은 ktlint에서 검사를 진행하지 않아 option + command + l 클릭을 해주면 좋을거 같아요.
감사합니다.

jiwon2724 committed:
[REFACTOR]#14 - snake case -> camel case 수정

myung6024 commented:
LGTM

myung6024 reviewed (approved):
LGTM

jiwon2724 committed:
[EDIT]#14 - drawable 컨벤션 수정

jiwon2724 committed:
[REFACTOR]#14 - MediumButton 컨벤션 수정

aurora32s reviewed (approved):
저도 LGTM👍🏻
